demo/addons/isochroneAddon/isochroneAddon.js

```diff
@@ -0,0 +1,334 @@
+      let element = e.currentTarget;
+      $(".selected.isochrone-mode").removeClass("selected");
+      $(element).addClass("selected");
+    }
+
+    /**
+     *switch between time and distance, display respective form and clears the form not displayed, adds/removes selected class 
+     * @param  {event} f
+     */
+
+    var switchModeParametre= function (f) {
+
+      let element = f.currentTarget;
+      $(".selected.parametre").removeClass("selected");
+      $(element).addClass("selected");
+      var parametreData = element.dataset.parametre;
+      if(parametreData === "temps"){
+        document.getElementById("temps").style.display = "block";
+        document.getElementById("distance").style.display = "none";
+        $('#distance :input').val('');
+      }else if(parametreData === "distance"){
+        document.getElementById("distance").style.display = "block";
+        document.getElementById("temps").style.display = "none";
+        $('#temps :input').val('');
+      }
+      
+    }
+       
+    /**
+     *gets forms inputs and sends infomation for isochrone to be calculate on click caculer button
+     */
+
+    var calcul = function () {
+            var times = [];
+            var distances = [];
+
+            /**
+             * Converts input values to correspond to time in seconds and distance in meters, adds all values to corresponding table
+             * @param  {input} "#heures_input", "#minutes_input", "#kilometres_input", "#metres_input"
+             * @param  {} .each(function(index
+             * @param  {number} input
+             */
+
+            $("#heures_input").each(function(index,input) {
+                if (input.value > 0) {
+                    times.push(input.value*3600);
+                };
+            });
+
+            $("#minutes_input").each(function(index,input) {
+                if (input.value > 0) {
+                    times.push(input.value*60);
+                };
+            });
+
+            $("#kilometres_input").each(function(index,input) {
+                if (input.value > 0) {
+                    distances.push(input.value*1000);
+                }
+            });
+
+            $("#metres_input").each(function(index,input) {
+                if (input.value > 0) {
+                    distances.push(input.value);
+                }
+            });
+
+            if (!_xy || (times.length === 0) && (distances.length === 0)) {
+                mviewer.alert("Isochrones : Il faut définir l'origine et au moins un temps de parcours ou une distance", "alert-info")
+                return;
+            }
+
+            /**
+             * adds all values from table together for one total time and total distance
+             * @param  {number} "totalTime", "totalDistance"
+             */
+
+            var totalTime = 0;
+            for(var t = 0; t<times.length; t++){
+              totalTime = totalTime + parseInt(times[t]);
+            }
```

Comment:
To get sum of array's values you could simplify this bloc by :

`var totalTime = times.reduce((a, b) => a + b, 0) // return 0 if !times.length`